lib/Util/License/Features/Subscribers.php

```diff
@@ -3,18 +3,37 @@
 namespace MailPoet\Util\License\Features;
 
 use MailPoet\Models\Subscriber as SubscriberModel;
-use MailPoet\Util\License\License;
+use MailPoet\Services\Bridge;
+use MailPoet\Settings\SettingsController;
 
 class Subscribers {
-  public $license;
-  const SUBSCRIBERS_LIMIT = 2000;
+  const SUBSCRIBERS_OLD_LIMIT = 2000;
+  const SUBSCRIBERS_NEW_LIMIT = 1000;
+  const NEW_LIMIT_DATE = '2019-11-00';
 
-  function __construct($license = false) {
-    $this->license = ($license) ? $license : License::getLicense();
+  private $license;
+
+  /** @var int */
+  private $installation_time;
+
+  /** @var int */
+  private $subscribers_count;
+
+  function __construct(SettingsController $settings) {
+    $has_mss_key = !empty($settings->get(Bridge::API_KEY_SETTING_NAME));
+    $has_premium_key = !empty($settings->get(Bridge::PREMIUM_KEY_SETTING_NAME));
```

Comment:
Please move this `$settings->get` and model calls from the constructor. We should use the constructor for passing dependencies. This code would access the database and this could be dangerous because DI container may instantiate this class before we safely check DB connection in Initializer.
You could just set $settings controller as private property and add `getLicence`. Settings are cached internally so accessing those is cheap. You could also cach `subscribers_count` later.
I also noticed that you added a repository `SubscribersRepository` with method `getTotalSubscribers`
Could you use this repository instead static call on old model?